### PR TITLE
(TWILL-179) Added support for custom ClassLoader for containers

### PR DESCRIPTION
**Pull request by chtyim**
- Added method TwillPreparer.setClassLoader
- Use system property "twill.custom.class.loader" to pass the class name of the custom ClassLoader

### CODE REVIEW

File: twill-yarn/src/test/java/org/apache/twill/yarn/CustomClassLoader.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  protected Class<?> findClass(String name) throws ClassNotFoundException {
+    if (!CustomClassLoaderRunnable.GENERATED_CLASS_NAME.equals(name)) {
+      return super.findClass(name);
+    }
+
+    // Generate a class that look like this:
+    //
+    // public class Generated {
+    //
+    //   public void announce(ServiceAnnouncer announcer, String serviceName, int port) {
```
Comment (anwar6953) [inline twill-yarn/src/test/java/org/apache/twill/yarn/CustomClassLoader.java]: Is the javadoc outdated/inconsistent?
`Cancellable` is declared as the return type, below on line 76.

Comment (chtyim) [inline twill-yarn/src/test/java/org/apache/twill/yarn/CustomClassLoader.java]: Will update.

**Comment by anwar6953** [comment]
1 minor comment; remainder LGTM.

File: twill-yarn/src/test/java/org/apache/twill/yarn/CustomClassLoader.java (hunk 1 above)
Comment (chtyim) [inline twill-yarn/src/test/java/org/apache/twill/yarn/CustomClassLoader.java]: actually the comment is correct. The generated method has `void` return type. The `announcer.announce` method has `Cancellable` return type, but the result gets ignored (hence the `mg.pop()` code in line 79).

**Pull request closed without merging**